test/puppetlabs/trapperkeeper/services/authorization/authorization_core_test.clj

```diff
@@ -144,7 +144,37 @@
                                        {:path "/"
                                         :type "path"
                                         :query-params {"env" [:notastring]}}
-                                       :allow "*"})))))
+                                       :allow "*"})))
+
+    (is (thrown-with-msg? IllegalArgumentException
+          #".* 'get', 'post', 'put', 'delete', 'head'"
+          (validate-auth-config-rule! {:match-request
+                                       {:path "/"
+                                        :type "path"
+                                        :method "gross"}
+                                       :allow "*"})))
+
+    (is (thrown-with-msg? IllegalArgumentException
+          #".* 'get', 'post', 'put', 'delete', 'head'"
+          (validate-auth-config-rule! {:match-request
+                                       {:path   "/"
+                                        :type   "path"
+                                        :method ["nasty" "gross"]}
+                                       :allow "*"})))
+
+    (is (not (nil? (validate-auth-config-rule!
+                     {:match-request
+                      {:path "/"
+                       :type "path"
+                       :method ["put" "get"]}
+                      :allow "*"}))))
+
+    (is (not (nil? (validate-auth-config-rule!
+                     {:match-request
+                             {:path "/"
+                              :type "path"
+                              :method ["put"]}
+                      :allow "*"}))))))
```

Comment:
I think it would be better to work the last two into the `valid-configs-pass` tests since they aren't invalid configs.